controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1428,6 +1428,11 @@ func (r *BareMetalHostReconciler) getHostFirmwareSettings(info *reconcileInfo) (
 	// Check if there are settings in the Spec that are different than the Status
 	if meta.IsStatusConditionTrue(hfs.Status.Conditions, string(metal3v1alpha1.FirmwareSettingsChangeDetected)) {
 
+		// Check if the status settings have been populated
+		if len(hfs.Status.Settings) == 0 {
+			return false, nil, errors.New("host firmware status settings not available")
```

Comment:
OK, this will prevent a transition out of Preparing in the case of, for example, the license not being set on certain hardware but if the user clears the Spec setting it will no longer be blocked.

Reply:
Yes. I extended TestHFSEmptyStatusSettings to include this scenario.